src/intervals.jl

```diff
@@ -0,0 +1,159 @@
+using .IntervalArithmetic
+
+function evaluate(a::Taylor1, dx::Interval)
+    dx == (-1..1) && return _evaluate(a, dx, Val(true))
+    dx == (0..1) && return _evaluate(a, dx, Val(false))
+    # Usual Horner rule
+    @inbounds begin
+        suma = a[end]*one(dx)
+        for k in a.order-1:-1:0
+            suma = suma*dx + a[k]
+        end
+    end
+    return suma
+end
+
+function _evaluate(a::Taylor1, dx::Interval, ::Val{true})
+    if iseven(a.order)
+        @inbounds begin
+            suma1 = a[end-1]
+            for k in a.order-3:-2:1
+                suma1 += a[k]
+            end
+            suma2 = a[end]
+            for k in a.order-2:-2:2
+                suma2 += a[k]
+            end
+        end
+    else
+        @inbounds begin
+            suma1 = a[end]
+            for k in a.order-2:-2:1
+                suma1 += a[k]
+            end
+            suma2 = a[end-1]
+            for k in a.order-3:-2:2
+                suma2 += a[k]
+            end
+        end
+    end
+    return a[0] + suma1*dx + suma2 * (one(dx) * (0..1))
+end
+
+function _evaluate(a::Taylor1, dx::Interval, ::Val{false})
+    @inbounds begin
+        suma = a[end]
+        for k in a.order-1:-1:1
+            suma = suma + a[k]
+        end
+    end
+    return a[0] + suma*dx
+end
+
+
+function evaluate(a::TaylorN, dx::IntervalBox{N,T}) where {N,T}
+
+    @assert N == get_numvars()
+    a_length = length(a)
+    suma = zero(constant_term(a)) + Interval{T}(0, 0)
+    @inbounds for homPol in length(a):-1:1
+        suma += evaluate(a.coeffs[homPol], dx)
+    end
+
+    return suma
+end
+
+function evaluate(a::HomogeneousPolynomial, dx::IntervalBox{N,T}) where {T, N}
+
+    @assert N == get_numvars()
+    dx == IntervalBox(-1..1, Val(N)) && return _evaluate(a, dx, Val(true))
```

Comment:
This check is probably quite slow. Can we record somewhere if the TaylorSeries or TaylorModel is normalized and call the specialised version?

Reply:
I agree with the observation; yet, I see no other way of branching to the specialized method. 

Regarding your question, for a `TaylorModelN` this is already in the struct; for a "general" `TaylorN` (or `HomogeneousPolynomial`) I see no point to include it.